src/Servers/HttpSys/src/NativeInterop/UrlGroup.cs

```diff
@@ -79,12 +79,13 @@ internal void RegisterPrefix(string uriPrefix, int contextId)
             {
                 if (statusCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_ALREADY_EXISTS)
                 {
-                    throw new HttpSysException((int)statusCode, string.Format(Resources.Exception_PrefixAlreadyRegistered, uriPrefix));
+                    throw new HttpSysException((int)statusCode, Resources.FormatException_PrefixAlreadyRegistered(uriPrefix));
                 }
-                else
+                if (statusCode == UnsafeNclNativeMethods.ErrorCodes.ERROR_ACCESS_DENIED)
                 {
-                    throw new HttpSysException((int)statusCode);
+                    throw new HttpSysException((int)statusCode, Resources.FormatException_AccessDenied(uriPrefix, Environment.UserName));
```

Comment:
Can we add a test for this?

Reply:
Is Environment.UserName in the right format to insert directly into the command?

Reply:
@halter73 I added a test in 8784857099fa085913ef4c4d03dd3e7743d1db30.
@Tratcher I included both the domain + user in `DOMAIN\User` format, as documented on [Configuring namespace reservations](https://docs.microsoft.com/en-us/dotnet/framework/wcf/feature-details/configuring-http-and-https#configuring-namespace-reservations), also in 8784857099fa085913ef4c4d03dd3e7743d1db30.